Engine/DataFeeds/LiveTradingDataFeed.cs

```diff
@@ -324,50 +329,56 @@ protected Subscription CreateDataSubscription(SubscriptionRequest request)
                                 (sender, args) => subscription.OnNewDataAvailable());
 
                             _exchange.AddDataHandler(request.Configuration.Symbol, data =>
-                            {
-                                var tick = data as Tick;
-
-                                if (tick?.TickType == TickType.OpenInterest && !tick.Suspicious)
                                 {
-                                    oiAggregator.ProcessData(tick);
-                                }
-                            });
+                                    var tick = data as Tick;
+
+                                    if (tick?.TickType == TickType.OpenInterest && !tick.Suspicious)
+                                    {
+                                        oiAggregator.ProcessData(tick);
+                                    }
+                                });
                             enumerator = oiAggregator;
                             break;
-                    }
-                }
-                else
-                {
-                    // tick subscriptions can pass right through
-                    var tickEnumerator = new EnqueueableEnumerator<BaseData>();
 
-                    _exchange.AddDataHandler(request.Configuration.Symbol, data =>
-                    {
-                        if (data.DataType == MarketDataType.Auxiliary)
-                        {
-                            tickEnumerator.Enqueue(data);
-                            subscription.OnNewDataAvailable();
-                        }
-                        else
-                        {
-                            var tick = data as Tick;
-                            if (tick?.TickType == request.Configuration.TickType)
-                            {
-                                tickEnumerator.Enqueue(data);
-                                subscription.OnNewDataAvailable();
-                                if (tick.TickType != TickType.OpenInterest)
+                        case nameof(Tick):
+                        default:
+                            // tick or streaming custom data subscriptions can pass right through
+                            var tickEnumerator = new EnqueueableEnumerator<BaseData>();
+
+                            _exchange.AddDataHandler(
+                                request.Configuration.Symbol,
```

Comment:
One thing to note is that for custom subscriptions, this `Symbol` instance might have an `underlying` which is **used for equality** (https://github.com/QuantConnect/Lean/blob/master/Common/SecurityIdentifier.cs#L754) and the incoming data might not have the underlying set, so we need to assert the `underlying` is set in the data producer or remove it here  🤔

Reply:
I think the data producer should be responsible for generating the correct symbol identifiers and setting the underlying symbol if required. If the symbol in the incoming data doesn't have the underlying set, the data handler will not be called and this is expected behavior, right?